tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -3223,8 +3223,8 @@ LogicalResult ConvertTFLCustomOp::matchAndRewrite(
     Operation* op, PatternRewriter& rewriter) const {
   auto tfl_custom_op = cast<TFL::CustomOp>(op);
   rewriter.replaceOpWithNewOp<tosa::CustomOp>(
-      op, op->getResultTypes(), tfl_custom_op.getCustomCode(), StringRef{},
-      StringRef{}, op->getOperands());
+      op, op->getResultTypes(), tfl_custom_op.getCustomCode(), rewriter.getStringAttr("TFL"),
+      tfl_custom_op.getCustomOption().cast<mlir::TFL::ConstBytesAttr>().getValue().str(), op->getOperands());
```

Comment:
And str() is required here as you need a StringAttr rather than ConstBytesAttr?

Reply:
That's right. I think ConstBytesAttr is a TFL dialect type, I went with a plain StringAttr for simplicity.
I thought about DictionaryAttr, and discussed it a bit with Sean after it merged upstream. From that discussion, there were attributes that would likely need an encoding even with DictionaryAttr. For our serialization format, we'd need to encode the dictionary, but that's a solvable problem.

We could reconsider if you've got a strong opinion. In this case, the Dictionary would just be the one string, but for other types of custom they could take advantage of the Dictionary.